components/PopDetails.js

```diff
@@ -0,0 +1,86 @@
+import styled from "styled-components";
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import Link from "next/link";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
+
+export default function PopDetails() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const { data, error } = useSWR(`/api/popstars/${id}`, fetcher);
+
+  if (error) {
+    return <h1>Nothing there..</h1>;
+  }
+
+  if (!data) {
+    return <h1>LOADING...</h1>;
+  }
+
+  async function handleAddTicket(event) {
+    event.preventDefault();
+
+    const ticketData = {
+      image: data.image,
+      rating: data.rating,
+      name: data.name,
+      location: data.location,
+      date: data.date,
+      price: data.price,
+      currency: data.currency,
+    };
+
+    let cartTickets = JSON.parse(localStorage.getItem("cartTicket"));
+    if (!Array.isArray(cartTickets)) {
+      cartTickets = [];
+    }
+
+    const existingTicket = cartTickets.find(
+      (ticket) => ticket.name === ticketData.name
+    );
+    if (existingTicket) {
+      console.log(`A ticket from '${data.name}' is already in the cart.`);
+      return;
+    }
+
+    cartTickets.push(ticketData);
+
+    localStorage.setItem("cartTicket", JSON.stringify(cartTickets));
+  }
```

Comment:
This function works great! But it's the same function as in the `RapPage.js`.
See if you can lift this function up and pass it down to both `PopDetails` and `RapDetails`. Doing this will result in less code because you don't have duplicate code and it's usually better to keep components as dumb as possible. Components shouldn't implement too much logic but receive it via props.